### PR TITLE
Supports multiple Noble objects

### PR DESCRIPTION
**Pull request by schnippe**
@sandeepmistry This changes supports multiple parallel noble instances with full backward compatibility.

You just use 

      var Noble = require('noble/lib/noble')
      var nobleA = new Noble ({NOBLE_HCI_DEVICE_ID : '0' })
      var nobleB = new Noble ({NOBLE_HCI_DEVICE_ID : '1' })

How do I keep the old commits out of this pull request?

### CODE REVIEW

**Comment by sandeepmistry** [comment]
@schnippe I typically sync my fork with upstream first, and then create a new branch per pull request I submit. See:

* https://help.github.com/articles/configuring-a-remote-for-a-fork/
* https://help.github.com/articles/syncing-a-fork/

Please close this pull request and submit a new one with the new changes only.

I have a few questions for you:
 * Why would someone want to use multiple noble instances in one app?
 * How would someone create more than one instance?

File: lib/linux/hci-ble.js [hunk 1]
```diff
@@ -4,12 +4,13 @@ var events = require('events');
...
-  var hciBle = __dirname + '/../../build/Release/hci-ble';
-
-  debug('hciBle = ' + hciBle);
+var HciBle = function(cfg) {
+  this._cfg = cfg
 
-  this._hciBle = spawn(hciBle);
+  var hciBle = __dirname + '/../../build/Release/hci-ble';
+  debug('hciBle = %s with hci%s', hciBle, this._cfg.NOBLE_HCI_DEVICE_ID);
+  var env = {NOBLE_HCI_DEVICE_ID : this._cfg.NOBLE_HCI_DEVICE_ID || '0' }
```
Comment (sandeepmistry) [inline lib/linux/hci-ble.js]: I don't think it should default to 0, leave undefined, and let ```hci_get_route``` in the C layer chose the default.

File: lib/linux/l2cap-ble.js [hunk 2]
```diff
@@ -196,7 +198,8 @@ L2capBle.prototype.connect = function() {
 
   debug(this._address + ': l2capBle = ' + l2capBle);
 
-  this._l2capBle = spawn(l2capBle, [this._address, this._addressType]);
+  var env = {NOBLE_HCI_DEVICE_ID : this._cfg.NOBLE_HCI_DEVICE_ID || '0' }
```
Comment (sandeepmistry) [inline lib/linux/l2cap-ble.js]: Same here, no default of 0

**Comment by schnippe** [comment]
Agreed, will do.

**Comment by schnippe** [comment]
@sandeepmistry 
`Why would someone want to use multiple noble instances in one app?`
I plan on putting multiple Raspberries and query them remotely, sometimes via Noble, sometimes the Raspberries just log for later use. Placing these Raspberries in every corner is sometimes not practical nor cost effective so I want to use multiple BLE-Thumbs on long USB cables going through walls and ceilings placing the actual receivers 10m apart. I do want to choose moving peripherals depended on signal strength so more Noble instances being in the same process would be helpful.     

`How would someone create more than one instance?`

      // on a Raspberries in RoomA
      var nobleA = new Noble ({NOBLE_HCI_DEVICE_ID : '0' })
      var nobleB = new Noble ({NOBLE_HCI_DEVICE_ID : '1' })

or on the main server:
 
      var nobleWsA = new Noble ({NOBLE_WEBSOCKET : 'ws://ble_room_a:8881/hw0' })
      var nobleWsB = new Noble ({NOBLE_WEBSOCKET : 'ws://ble_room_a:8881/hw1' })

I am preparing matching pull requests.

**Comment by sandeepmistry** [comment]
@schnippe thanks,

It would be great if you also included a "advanced usage on Linux" mark down file, that contained the above and any other useful info.

One more question: How would you require ``Noble`` instead the current ```var noble = require('noble');```

**Comment by schnippe** [comment]
@sandeepmistry You would have to do:
 
     var Noble = require('noble/lib/noble')
     var adapters = [
       new Noble ({NOBLE_HCI_DEVICE_ID : '0' }), 
       new Noble ({NOBLE_HCI_DEVICE_ID : '1' })
     ]

Should we make it like this? : 

     var Noble = require('noble/with_configuration')
     var adapters = [
       new Noble ({NOBLE_HCI_DEVICE_ID : '0' }), 
       new Noble ({NOBLE_HCI_DEVICE_ID : '1' })
     ]

Looks cleaner but is basically just a shim.

**Comment by sandeepmistry** [comment]
@schnippe thanks,

How about this?

```javascript
var Noble = require('noble/Noble');
```

**Pull request closed without merging**